src/mixins/ui/lasso-shapes/LatLonViewIntersectUtils.js

```diff
@@ -36,6 +37,289 @@ function canIntersect(numerator, denominator) {
+        console.assert(false, `Collinear intersection needs completing`)
+
+        return true
+      }
+      const numerator_t = numerator_t_functor()
+      if (
+        canIntersect(numerator_t, denominator_t) &&
+        canIntersect(numerator_u_functor(), denominator_u_functor())
+      ) {
+        // NOTE: denominator_t should never be zero. There's an early out for
+        // that case.
+        const t = numerator_t / denominator_t
+
+        if (MathExt.floatingPtEquals(t, current_t)) {
+          // already found this specific interesection, so don't re-add
+          // In this specific use case, since we're doing intersection tests
+          // against each wall of an axis-aligned bounding box, the original
+          // line-segment can intersect two walls that meet at a corner. In
+          // that case one of the endpoitns of the original line defines the
+          // bounds at that corner. That can result in recalculating the same
+          // t at the corner for the two separate walls, so avoiding adding
+          // a duplicate intersection point with this test.
+          return false
+        }
+
+        let lonlat_pt = null
+        let screen_pt = null
+        if (MathExt.floatingPtEquals(t, 0)) {
+          // the intersection point is the start point of the original segment, so
+          // just re-use the start coord
+          lonlat_pt = start_point_data.lonlat_point
+          screen_pt = start_point_data.screen_point
+        } else if (MathExt.floatingPtEquals(t, 1)) {
+          // the intersection point is the end point of the original segment, so
+          // just re-use the end coord
+          lonlat_pt = end_point_data.lonlat_point
+          screen_pt = end_point_data.screen_point
+        } else {
+          // calculate the lon/lat and screen point for the
+          // intersection point given t
+          const new_lonlat_pt = Point2d.create(-delta_x * t, -delta_y * t)
+          Point2d.addVec2(
+            new_lonlat_pt,
+            start_point_data.lonlat_point,
+            new_lonlat_pt
+          )
+          lonlat_pt = new_lonlat_pt
+
+          const new_screen_pt = Point2d.clone(new_lonlat_pt)
+          // conver the new segment point back to mercator for drawing
+          LatLonUtils.conv4326To900913(new_screen_pt, new_screen_pt)
+
+          // now convert to screen space
+          Point2d.transformMat2d(
+            new_screen_pt,
+            new_screen_pt,
+            world_to_screen_matrix
+          )
+          screen_pt = new_screen_pt
+        }
+        rtn_obj.subdivide = true
+
+        // insersection points must be stored by t in ascending order.
+        // There will be exactly 2 points in the end, so just need to check
+        // whether to insert this new intersection point at the front or back
+        // of the returned point array
+        if (t < current_t) {
+          rtn_obj.lonlat_pts.splice(0, 0, lonlat_pt)
```

Comment:
You should be able to use `unshift` here instead of splice to add to the beginning of the array